_posts/2024-10-19-openloco-v24.10.md

```diff
@@ -0,0 +1,150 @@
+    image_path: https://github.com/user-attachments/assets/5dec9fe5-4e24-4162-9410-2a715080992a
+    alt: "In-game charts (before)"
+    title: "In-game charts (before)"
+  - url: https://github.com/user-attachments/assets/db8755b8-375a-46b8-ba3c-8bfc8f492de2
+    image_path: https://github.com/user-attachments/assets/db8755b8-375a-46b8-ba3c-8bfc8f492de2
+    alt: "In-game charts (after)"
+    title: "In-game charts (after)"
+---
+
+OpenLoco v24.10 is out! In this release we've added support for subfolders in
+the object index, reimplemented chart drawing, started work on a new widget system,
+and many other things!
+
+For a complete summary of changes, please find the
+[OpenLoco changelog](https://github.com/OpenLoco/OpenLoco/releases/tag/v24.10) on GitHub.
+
+## First off: assorted bug fixes (#2623, #2625, #2653, #2658)
+
+As usual, we start off with some bug fixes. While we're taking great care not to break the game
+as we're reimplementing bits of it, we're only human, and do make mistakes. Often, it's
+small assumptions that end up not holding.
+
+First on the list is a crash that could occur when attempting to place a track underground.
+Under some circumstances, this could lead to a negative Z coordinate, which is invalid in the game's
+coordinate system. Oops!
+
+Second is a crash that could occur when recalculting a vehicle's reliability. As it turns out,
+this is hard to do if a vehicle doesn't have any cars or parts last, such as when you're selling
+it gradually. It's a small code change, but selling no longer causes a crash now.
+
+We continue with a more obscure crash: resizing the game window while a loading bar is open
+could cause the game to crash! This needed a bit more shuffling of code, but in the end, the game
+no longer crashes! As an added bonus, the loading bar is now repositioned when the game window
+is resized. Nice!
+
+Finally, and perhaps most impressively, some track rendering bugs were fixed. In previous reports,
+we've detailed how the track 'paint' functions have been reimplemented. It turns out there were a
+few unintuitive mistakes in bounding box calculations where track and track additions were concerned.
+In some edge cases, entire vehicles could temporarily clip out of existence! We include an extreme
+example below, and what it's supposed to look like.
+
+{% include gallery id="gallery_track_clipping" %}
+
+## Updates to object index and selection (#2629, #2650)
+
+A frequent request we've gotten over the years, is to allow the game object data to be organised in a more
+flexible way. In order for this to be possible, the full object manager and index code needed to be
+reimplemented. Good news, we're at that point! This month, @duncanspumpkin reworked the object index code,
+such that the index process now recurses into subfolders. Moreover, you can now put object data files
+in the OpenLoco folder, as well as your OpenLoco user folder. This way, you can keep your custom objects
```

Comment:
How exactly does this work. Can you plop a new 'ObjData' folder in this folder and it'll just work? Does it need to be a specific named folder or any will do?

Reply:
Ah, good question. While `ObjData` is what Locomotion uses, in both the user folder and main OpenLoco folder, it will look for a folder called `objects`. I'll edit to clarify.